packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -45,36 +46,31 @@ contract L2OutputOracle is Initializable, Semver {
     uint256 public startingTimestamp;
 
     /**
-     * @notice The number of the most recent L2 block recorded in this contract.
+     * @notice Array of L2 output proposals.
      */
-    uint256 public latestBlockNumber;
-
-    /**
-     * @notice A mapping from L2 block numbers to the respective output root. Note that these
-     *         outputs should not be considered finalized until the finalization period (as defined
-     *         in the Optimism Portal) has passed.
-     */
-    mapping(uint256 => Types.OutputProposal) internal l2Outputs;
+    Types.OutputProposal[] internal l2Outputs;
 
     /**
      * @notice Emitted when an output is proposed.
      *
      * @param outputRoot    The output root.
-     * @param l1Timestamp   The L1 timestamp when proposed.
+     * @param l2OutputIndex The index of the output in the l2Outputs array.
      * @param l2BlockNumber The L2 block number of the output root.
+     * @param l1Timestamp   The L1 timestamp when proposed.
      */
     event OutputProposed(
         bytes32 indexed outputRoot,
-        uint256 indexed l1Timestamp,
-        uint256 indexed l2BlockNumber
+        uint256 indexed l2OutputIndex,
+        uint256 indexed l2BlockNumber,
+        uint256 l1Timestamp
     );
 
     /**
      * @notice Emitted when outputs are deleted.
      *
-     * @param l2BlockNumber First L2 block number deleted.
+     * @param l2OutputIndex First L2 output index deleted.
      */
-    event OutputsDeleted(uint256 indexed l2BlockNumber);
+    event OutputsDeleted(uint256 indexed l2OutputIndex);
```

Comment:
I think OutputDeleted would be more descriptive now.
Or keep the name, and make it
`OutputsDeleted(uint256 indexed l2OutputIndex, uint256 numOutputsDeleted);

Reply:
from the pov of an offchain actor, getting `numOutputsDeleted` feels a bit less helpful than say getting the new height because getting the num outputs deleted requires context (the total length) to know what the new height is

Reply:
Did a different change but same idea